### DISCUSSION TITLE
Is there a way to control an outlines visibility from different angles?

### DISCUSSION THREAD

**Discussion by MeganDrawsaLot** [bnpr/Malt#49] [Q&A] [answered]
This might be kind of a crazy question, but is there a way to add or remove outlines on the same mesh in Malt?
Or adjust the line depth and normal thresh holds in different places on the mesh separately?

What I'm trying to achieve is to see a specific line from one angle but have it fade/vanish when it's viewed at another angle.

Hope this makes sense, my brain has turned to mush a bit as I try to figure it out.  I thought doing something with data transfer might work, though I am not very familiar with the modifier and how to use it.  I got something close to what I wanted when I made a second mesh and used smooth + project face interpolated on the original mesh, but lost some of the outlines I wanted to keep.

I added a pic to help illustrate what I've tried/am trying to do.

Thanks in advance to anyone who has suggestions. :)

![Illustration](https://user-images.githubusercontent.com/77802440/105399178-4652a780-5bf1-11eb-9b10-26f6ad836c06.png)

**Comment by pragma37** [comment, marked as answer]
Yes, you could drive the thresholds, the line width and/or the line color with textures or vertex colors.

But you need to make your own shader for that. Are you familiar with GLSL?

I think the easiest way for your use case would be just scaling the line width with a vertex color.
Start with the advanced_line example and change the line that says:
`PO.line_width = line;`
to:
`PO.line_width = line * S.color[0].r;`

Now create a vertex color layer for your mesh. The white parts will render the line as previously and the black ones won't have lines.

**Reply by MeganDrawsaLot** [reply]
I actually don't know all that much about coding and had to look up what GLSL is.  But I am willing to learn if it helps get the results I need.  I will look up how to make a vertex color layer and look at your "From Nodes To Code" tutorial. :)  
If you know of any resources I could use to help me out more I'd really appreciate them.

Thank you for taking the time to answer my questions. :D
